com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/MathUtil.java

```diff
@@ -46,7 +48,28 @@ public static Vector3d ddfToVecmath(Vector3One v) {
         return new Vector3d(v.getX(), v.getY(), v.getZ());
     }
 
-    public static Quat4d ddfToVecmath(Quat q) {
+    public static boolean isValid(Quat q) {
+        if (q.getX() == 0  && q.getY() == 0 && q.getZ() == 0 && q.getW() == 0) {
+            return false;
+        }
+        if (Float.isNaN(q.getX()) || Float.isNaN(q.getY()) || Float.isNaN(q.getZ()) || Float.isNaN(q.getW())) {
+            return false;
+        }
+        if (Float.isInfinite(q.getX()) || Float.isInfinite(q.getY()) || Float.isInfinite(q.getZ()) || Float.isInfinite(q.getW())) {
+            return false;
+        }
+        float lenSq = q.getX()*q.getX() + q.getY()*q.getY() + q.getZ()*q.getZ() + q.getW()*q.getW();
+        if (Math.abs(lenSq - 1.0f) > 1e-4f) {
+            return false;
+        }
+        return true;
+    }
+
+    public static Quat4d ddfToVecmath(Quat q, String owner) {
+        if (!isValid(q)) {
+            String err = new StringBuilder().append("Invalid quaternion: ").append(q).append(" in ").append(owner).toString();
+            throw new RuntimeException(new CompileExceptionError(err));
```

Comment:
`CompileExceptionError` wrapped in `RuntimeException` works good in both the editor and in bob